example/multi_fem/LinAlgBLAS.hpp

```diff
@@ -103,17 +103,17 @@ namespace Kokkos {
 
 template< typename ScalarX /* Allow mix of const and non-const */ ,
           typename ScalarY /* Allow mix of const and non-const */ ,
-          class L , class D ,
-          class MX /* Allow any management type */ ,
-          class MY /* Allow any management type */ >
+          class D>
 inline
 double dot( const size_t n ,
-            const View< ScalarX * , L , D , MX > & x ,
-            const View< ScalarY * , L , D , MY > & y ,
+            const View< ScalarX * , D > & x ,
+            const View< ScalarY * , D > & y ,
             comm::Machine machine )
 {
   double global_result = 0 ;
   double local_result = 0 ;
+  using x_type = View<ScalarX*, D>;
```

Comment:
Nope, this will return a reference type. `typename std::decay<decltype(x)>::type` would be the most generally robust, but this is clearer for people who are looking at this as just an example. `LinAlgBLAS.hpp` isn't supposed to be used outside this example.